app/src/main/java/org/oppia/android/app/devoptions/platformparameters/PlatformParameterItemViewModel.kt

```diff
@@ -0,0 +1,101 @@
+package org.oppia.android.app.devoptions.platformparameters
+
+import androidx.annotation.ColorInt
+import androidx.databinding.ObservableField
+import org.oppia.android.app.model.PlatformParameterId
+import org.oppia.android.app.model.PlatformParameterValue
+import org.oppia.android.app.model.SyncStatus
+import org.oppia.android.app.translation.AppLanguageResourceHandler
+import org.oppia.android.app.view.models.R
+import org.oppia.android.app.viewmodel.ObservableViewModel
+import org.oppia.android.util.locale.OppiaLocale
+
+/**
+ * [ViewModel] for displaying a platform parameter in the RecyclerView of
+ * [PlatformParametersFragment].
+ */
+class PlatformParameterItemViewModel(
+  val platformParameterId: PlatformParameterId,
+  val currentValue: PlatformParameterValue,
+  val syncStatus: SyncStatus,
+  private val machineLocale: OppiaLocale.MachineLocale,
+  private val resourceHandler: AppLanguageResourceHandler
+) : ObservableViewModel() {
+  /** The observable boolean value of the parameter, used for switch toggles in UI. */
+  val isChecked = ObservableField(currentValue.boolean)
+
+  /** Whether the input field should be visible (i.e., if the value is not a boolean). */
+  val isInputVisible = ObservableField(!currentValue.hasBoolean())
+
+  /** The display name of the platform parameter. */
+  val platformParameterDisplayText = ObservableField(getPlatformParameterDisplayName())
+
+  /** The sync status text for the platform parameter. */
+  val syncStatusDisplayText = ObservableField(getSyncStatusText())
+
+  /** The user-editable value of the platform parameter (if it is a string or integer). */
+  val inputValue = ObservableField(
+    if (currentValue.hasString()) currentValue.string
+    else if (currentValue.hasInteger()) currentValue.integer.toString()
+    else ""
+  )
+
+  /**
+   * Callback invoked when a boolean-type platform parameter is toggled.
+   * Passes the parameter ID and the new value.
```

Comment:
Aligning with the other KDoc.
```suggestion
   * Passes the parameter ID and the new boolean value.
```

Reply:
updated